### ISSUE TITLE
null.{isInstanceOf, asInstanceOf} semantics differs 

### ISSUE THREAD

**Issue by densh** [scala-native/scala-native#283]
Currently all operations on `null`s are undefined behaviour. This is not consistent with `isInstanceOf` and `asInstanceOf` having a defined non-throwing semantics on null values on the JVM.

**Comment by pbatko** [comment]
I've started hacking on the solution here[1] and I it looks like `null.instanceOf[T]` case started working .
I have a few questions:
1. Is `ClassLowering` a good place for implementing it?
2. `scala.scalanative.compiler.pass.ClassLowering#preInst` returns `Seq[Inst]`  so I created a dummy comparison instead of  a constant value. How can I make it work with a constant?
3. Should I create a PR already, even though my code is still WIP? I think a PR would be a good place to ask further questions.

[1]
https://github.com/pbatko/scala-native/commit/9d08b5ddd0593a7233dc31f89221d94eb2ebf636#diff-e36ca1bc7cfe963d5f07664423e87910R104
